client/src/__tests__/App.test.tsx

```diff
@@ -4,8 +4,9 @@ import api from '~/api';
 
 jest.mock('~/api');
 
-test('App renders without exceptions', async () => {
+test('App renders without exceptions', async done => {
   await act(async () => {
-    render(<App api={api} />);
+    await render(<App api={api} />);
```

Comment:
as far as I understand those functions are not async and to be honest, they should not to be that
any async action inside is either: timer-based or io-based; and both cases can be handled with
[timer mocks](https://jestjs.io/docs/en/timer-mocks) or [waitFor](https://testing-library.com/docs/dom-testing-library/api-async)

also, there is no need to use done callback while using async test, just return promise from test:
https://jestjs.io/docs/en/asynchronous#promises